fictadvisor-web/src/components/pages/admin/disciplines-admin-page/pages/disciplines-edit-page/components/disciplines-info-section/DisciplinesInfoSection.tsx

```diff
@@ -0,0 +1,348 @@
+'use client';
+import { type FC, useEffect, useRef, useState } from 'react';
+import { useQuery } from 'react-query';
+import { PlusIcon, TrashIcon } from '@heroicons/react/24/outline';
+import { Box, Typography } from '@mui/material';
+import { isAxiosError } from 'axios';
+
+import Button from '@/components/common/ui/button-mui';
+import {
+  ButtonSize,
+  ButtonVariant,
+} from '@/components/common/ui/button-mui/types';
+import Divider from '@/components/common/ui/divider';
+import { Dropdown } from '@/components/common/ui/form';
+import { FieldSize } from '@/components/common/ui/form/common/types';
+import { DropDownOption } from '@/components/common/ui/form/dropdown/types';
+import {
+  IconButton,
+  IconButtonColor,
+  IconButtonShape,
+} from '@/components/common/ui/icon-button';
+import { useToastError } from '@/hooks/use-toast-error/useToastError';
+import DatesAPI from '@/lib/api/dates/DatesAPI';
+import DisciplineAPI from '@/lib/api/discipline/DisciplineAPI';
+import AddDiscipline from '@/lib/api/discipline/types/AddDiscipline';
+import GetAllDisciplineTeachers from '@/lib/api/discipline/types/GetAllDisciplineTeachers';
+import GroupAPI from '@/lib/api/group/GroupAPI';
+import SubjectAPI from '@/lib/api/subject/SubjectAPI';
+import TeacherAPI from '@/lib/api/teacher/TeacherAPI';
+import { AdminDiscipline } from '@/types/discipline';
+import { DisciplineTeacher, TeacherRole } from '@/types/teacher';
+
+import { disciplineTeacherInitialValues } from '../../../../components/disciplines-admin-search/constants';
+
+import * as styles from './DisciplinesInfoSection.styles';
+
+interface DisciplinesInfoSectionProps {
+  discipline: AdminDiscipline;
+  onEditSubmit: (values: AddDiscipline) => void;
+}
+
+const rolesData = ['LECTURER', 'LABORANT', 'PRACTICIAN', 'EXAMINER', 'OTHER'];
+
+const DisciplinesInfoSection: FC<DisciplinesInfoSectionProps> = ({
+  discipline,
+  onEditSubmit,
+}) => {
+  const [semesterId, setSemesterId] = useState<string>();
+  const [groupId, setGroupId] = useState<string>();
+  const [subjectId, setSubjectId] = useState<string>();
+  const [disciplineTeachers, setDisciplineTeachers] =
+    useState<GetAllDisciplineTeachers>();
+  const groups = useRef<DropDownOption[]>();
+  const semesters = useRef<DropDownOption[]>();
+  const teachers = useRef<DropDownOption[]>();
+  const subjects = useRef<DropDownOption[]>();
+  const toastError = useToastError();
+
+  useEffect(() => {
+    onEditSubmit({
+      groupId: groupId as string,
```

Comment:
you don't need to hardcode groupId: groupId if you have similar names, write only groupId instead it